### DISCUSSION TITLE
What's the difference between version and constraints.version in routes options?

### DISCUSSION THREAD

**Discussion by ligreman** [fastify/fastify#5399] [Q&A] [answered]
I'm learning Fastify and now I'm trying the api versioning feature. In the docs there are two options for the routes that seems the same thing to me:
- **version**: a [semver](https://semver.org/) compatible string that defined the version of the endpoint. [Example](https://fastify.dev/docs/latest/Reference/Routes/#version-constraints).

- constraints: defines route restrictions based on request properties or values, enabling customized matching using [find-my-way](https://github.com/delvedor/find-my-way) constraints. Includes built-in **version** and host constraints, with support for custom constraint strategies.

https://fastify.dev/docs/latest/Reference/Routes/#routes-options

The example in the "version" one even redirects to the constraints section of the docs. Are these options the same? is version a shortcut of constraints.version?

I've tested both and aparently do the same thing... what's the difference doing:
`
fastify.route({
            method: 'GET',
            url: '/foo',
            constraints: {version: '1.1.0'}
        });
`
and
`
fastify.route({
            method: 'GET',
            url: '/foo',
            version: '1.1.0'
        });
`

Thanks!

**Comment by metcoder95** [comment, marked as answer]
They are the same, and internally the `version` property is added as `constraints.version` if passed while adding the route.
This is currently under a deprecation notice and most likely will be drop in the next major. Its recommended to use `constraints: { version: 'x.x.x' }` instead
